### ISSUE TITLE
Make Dolt work with mydumper

### ISSUE THREAD

**Issue by timsehn** [dolthub/dolt#8592]
https://github.com/mydumper/mydumper

A user is trying to load a database with myloader that he created with mydumper. It fails immediately in Dolt with:

```
** (myloader:736152): WARNING **: 22:23:15.511: Connection 5 - ERROR 1105: could not resolve initial root for database dbname/main
```

The database `dbname` is created but nothing else is imported.

**Comment by timsehn** [comment]
First bug:
```
$ mydumper -h dolthub-us-jails-6.dbs.hosted.doltdb.com -P 3306 -u <redacted> -p <redacted>

** (mydumper:31805): CRITICAL **: 12:17:51.572: Error executing SHOW SLAVE STATUS - ERROR 1105: syntax error at position 11 near 'SLAVE'
```

**Comment by timsehn** [comment]
`SHOW SLAVE` is deprecated and is being replaced by `SHOW REPLICA`.

**Comment by timsehn** [comment]
Made a bug in mydumper:

https://github.com/mydumper/mydumper/issues/1680

But we may just have to support `SHOW SLAVE STATUS`.

**Comment by jycor** [comment]
PRs supporting `SHOW SLAVE STATUS`:
- https://github.com/dolthub/dolt/pull/8600
- https://github.com/dolthub/go-mysql-server/pull/2767
- https://github.com/dolthub/vitess/pull/381

**Comment by jycor** [comment]
PR supporting `NO_AUTO_CREATE_USER` sql_mode option:
- https://github.com/dolthub/go-mysql-server/pull/2769

**Comment by timsehn** [comment]
Now mydumper fails with
```
 11:12:55.158593 -0800 PST m=+3.148194917" connectionID=4 query="START TRANSACTION /*!40108 WITH CONSISTENT SNAPSHOT */"
```

**Comment by timsehn** [comment]
Now myloader fails on:
```sql
CREATE TABLE `help_topic` ( `help_topic_id` int(10) unsigned NOT NULL, `name` char(64) NOT NULL, `help_category_id` smallint(5) unsigned NOT NULL, `description` text NOT NULL, `example` text NOT NULL, `url` text NOT NULL, PRIMARY KEY (`help_topic_id`), UNIQUE KEY `name` (`name`) ) ENGINE=Aria DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci PAGE_CHECKSUM=1 TRANSACTIONAL=0 COMMENT='help topics'"
```

Error is:
```
WARN[0069] error running query                           connectTime="2024-11-26 11:17:48.130184 -0800 PST m=+69.448642210" connectionDb=mysql connectionID=4 error="syntax error at position 375 near 'PAGE_CHECKSUM'" query="CREATE TABLE `help_topic` ( `help_topic_id` int(10) unsigned NOT NULL, `name` char(64) NOT NULL, `help_category_id` smallint(5) unsigned NOT NULL, `description` text NOT NULL, `example` text NOT NULL, `url` text NOT NULL, PRIMARY KEY (`help_topic_id`), UNIQUE KEY `name` (`name`) ) ENGINE=Aria DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci PAGE_CHECKSUM=1 TRANSACTIONAL=0 COMMENT='help topics'"
```

I think we need to support `PAGE_CHECKSUM` and `TRANSACTIONAL` there.

**Comment by jycor** [comment]
We are missing MariaDB specific collations, namely `utf8mb4_uca1400_ai_ci`
[MariaDB Docs](https://mariadb.com/kb/en/supported-character-sets-and-collations/#:~:text=1270%20%7C%20%20%20%20%20%20%20%20%20%7C%20Yes%20%20%20%20%20%20%7C%20%20%20%20%20%20%208%20%7C%0A%7C-,uca1400_ai_ci,-%7C%20NULL%20%20%20%20%20%7C%20NULL%20%7C%20NULL)

**Comment by CodeBleu** [comment]
Figured I'd link this here too :smile: 
https://github.com/dolthub/vitess/pull/381#issuecomment-2508186078

**Comment by jycor** [comment]
Conservatively bumping the reported MySQL version from `8.0.11` to `8.0.23` removed all errors relating to deprecated `master/slave` syntax: https://github.com/dolthub/go-mysql-server/pull/2776

**Comment by jycor** [comment]
We are missing support for `STOP REPLICA IO_THREAD` and `STOP REPLICA SQL_THREAD`.

[MySQL Docs](https://dev.mysql.com/doc/refman/8.4/en/stop-replica.html)

**Comment by jycor** [comment]
We need to match MySQL's `EXPLAIN` output:
https://github.com/dolthub/go-mysql-server/pull/2784

**Comment by jycor** [comment]
Hey @CodeBleu, I believe we've reached a point where a Mydumper and Myloader actually works with dolt now!
At least a very basic dump and import seem to be working. 

There are a few caveats and you may still see some errors, but it should work
More information in this blog: 
https://www.dolthub.com/blog/2024-12-13-dolt-mydumper/

Please let us know how it works for you!

**Comment by CodeBleu** [comment]
@timsehn @jycor 
I'm starting to test things again with trying to restore my large DB into Dolt via myloader.  I have updated both mydumper/myloader and dolt to the latest versions and attempted to run the import with myloader and still get this:

 (myloader:1487968): WARNING : 14:38:06.717: Failed to execute statement: could not resolve initial root for database dbname/main (1105)

I don't understand where the /main is coming from and if that has anything to do with it.

I have not started fresh with dolt, I only stopped my systemd service I have for the doltdb, then ran the curl command to pull down and run the install.sh to upgrade dolt and then started dolt again.  I did do a drop database dbname and I ran the CALL command to purge dropped.  This is when I tried to import it with myloader.
